zrml/swaps/src/tests.rs

```diff
@@ -3015,6 +3019,225 @@ fn pool_exit_with_exact_asset_amount_fails_if_liquidity_drops_too_low() {
+        let pool_id = 0;
+
+        assert_ok!(Swaps::pool_join_with_exact_pool_amount(
+            Origin::signed(ALICE),
+            pool_id,
+            ASSET_A,
+            _2,
+            u128::MAX,
+        ));
+        assert!(PoolsCachedForArbitrage::<Runtime>::contains_key(pool_id));
+        PoolsCachedForArbitrage::<Runtime>::remove(pool_id);
+
+        assert_ok!(Swaps::pool_join_with_exact_asset_amount(
+            Origin::signed(ALICE),
+            pool_id,
+            ASSET_A,
+            _2,
+            0,
+        ));
+        assert!(PoolsCachedForArbitrage::<Runtime>::contains_key(pool_id));
+        PoolsCachedForArbitrage::<Runtime>::remove(pool_id);
+
+        assert_ok!(Swaps::pool_exit_with_exact_asset_amount(
+            Origin::signed(ALICE),
+            pool_id,
+            ASSET_A,
+            _1,
+            u128::MAX,
+        ));
+        assert!(PoolsCachedForArbitrage::<Runtime>::contains_key(pool_id));
+        PoolsCachedForArbitrage::<Runtime>::remove(pool_id);
+
+        assert_ok!(Swaps::pool_exit_with_exact_pool_amount(
+            Origin::signed(ALICE),
+            pool_id,
+            ASSET_A,
+            _1,
+            0,
+        ));
+        assert!(PoolsCachedForArbitrage::<Runtime>::contains_key(pool_id));
+        PoolsCachedForArbitrage::<Runtime>::remove(pool_id);
+
+        assert_ok!(Swaps::swap_exact_amount_in(
+            Origin::signed(ALICE),
+            pool_id,
+            ASSET_A,
+            _1,
+            ASSET_B,
+            Some(0),
+            None,
+        ));
+        assert!(PoolsCachedForArbitrage::<Runtime>::contains_key(pool_id));
+        PoolsCachedForArbitrage::<Runtime>::remove(pool_id);
+
+        assert_ok!(Swaps::swap_exact_amount_out(
+            Origin::signed(ALICE),
+            pool_id,
+            ASSET_A,
+            Some(u128::MAX),
+            ASSET_B,
+            _1,
+            None,
+        ));
+        assert!(PoolsCachedForArbitrage::<Runtime>::contains_key(pool_id));
+    });
+}
+
+#[test]
+fn on_idle_skips_arbitrage_if_price_does_not_exceed_threshold() {
+    ExtBuilder::default().build().execute_with(|| {
+        frame_system::Pallet::<Runtime>::set_block_number(1);
+        let assets = ASSETS;
+        assets.iter().cloned().for_each(|asset| {
+            assert_ok!(Currencies::deposit(asset, &BOB, _10000));
+        });
+        assert_ok!(Swaps::create_pool(
+            BOB,
+            assets.into(),
+            ASSET_A,
+            0,
+            ScoringRule::CPMM,
+            Some(0),
+            Some(<Runtime as crate::Config>::MinLiquidity::get()),
+            Some(vec![_3, _1, _1, _1]),
+        ));
+        let pool_id = 0;
+        // Force the pool into the cache.
+        crate::PoolsCachedForArbitrage::<Runtime>::insert(pool_id, ());
+        Swaps::on_idle(System::block_number(), Weight::max_value());
+        System::assert_has_event(Event::ArbitrageSkipped(pool_id).into());
+    });
+}
```

Comment:
Maybe a comment for the total spot price calculation? That everyone can see, that the arbitrage amount does not exceed arbitrage threshold?

Reply:
I see, you create no imbalance like in the test below. So the total spot price should be 1 ZTG, right? But a comment about the calculation would be great.

Reply:
799e42d5b1849a74e3b15ca1735bf76dcffff280